rewrite-maven/src/main/java/org/openrewrite/maven/NormalizeMavenVariables.java

```diff
@@ -0,0 +1,52 @@
+package org.openrewrite.maven;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.maven.MavenIsoVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class NormalizeMavenVariables extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Normalize Maven variables";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Variables are all referenced by the prefix `project.`. You may also see references with `pom.` as the " +
+               "prefix, or the prefix omitted entirely - these forms are now deprecated and should not be used.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenIsoVisitor<ExecutionContext>() {
+            private final List<String> properties = Arrays.asList(
+                    "basedir",
+                    "groupId",
+                    "artifactId",
+                    "version",
+                    "build.timestamp"
+            );
+
+            @Override
+            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext executionContext) {
+                Xml.Tag t = super.visitTag(tag, executionContext);
+                Optional<String> value = t.getValue();
+                if (value.isPresent()) {
+                    String newValue = value
+                            .filter(v -> properties.stream().anyMatch(prop -> v.equals("${" + prop + "}")))
```

Comment:
IMO for this recipe to be complete, more replacements should be done:
- replace content inside tagValue. e.g. `<name>${artifactId} :: ${version}</name>` should be changed to `<name>${project.artifactId} :: ${project.version}</name>`
- replace `pom.` with `project.`, e.g. `${pom.version}` with `${project.version}`